File: pset1/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{

    int height;
    int no_of_hash;
    int no_of_space;
    
    //set variables that will be used for building the pyramid.

    do
    {
        printf("Height: ");
        height = get_int();
    }
    while (height < 0 || height > 23);
    
    //prompt for height of pyramind, reject <0 or >= 24
    
    for (int i = 0; i < height; i++)
    {
        no_of_hash = 2 + i;
        no_of_space = height - 1 - i;
        int space_left = no_of_space;
        int hash_left = no_of_hash;
        
        //establish a for loop that grows the no. of hashes and shrinks the no. of spaces of each row
        
        while (space_left > 0)
        {
            printf(" ");
            space_left--;
        }
        
        //print space until no. of spaces = "height - 1 - i"
        
        while (hash_left > 0)
        {
            printf("#");
            hash_left--;
            
        //print hashes until no. of hashes =  "2 + i"    
      
        }
        printf("\n");
    }
}

File: pset6/mario/less/mario.py
import cs50

height = 0

while height <= 0 or height >= 24:
    height = cs50.get_int("Height: ")

for i in range(height):
    no_of_hash = 2 + i
    no_of_space = height - 1 - i
    space_left = no_of_space
    hash_left = no_of_hash

    while space_left > 0:
        print(" ", end="")
        space_left -= 1

    while hash_left > 0:
        print("#", end="")
        hash_left -= 1

    print('')
